doc/decisions/DR024-Versioning-traits-and-specifications-method.md

```diff
@@ -0,0 +1,365 @@
+
+This DR attempts to narrow the focus onto a single point of contention
+that became a recurring theme, whether OpenAssetIO trait and
+specification versioning information should be communicated as part of
+the data, or agreed upon as a secondary "handshake" interaction between
+host and manager. For the purposes of this DR, we will refer to these
+two categories as "in-data" and "out-of-data".
+
+In attempting to keep its scope limited, this DR does not propose a
+specific solution, but merely provides representative examples of
+what a solution in each proposed paradigm might look like. It is generally
+assumed further facilitative work (for example, convenience utilities
+in `traitgen` and `mediacreation` will be decided upon and implemented
+at a later date.)
+
+## Relevant data
+
+Over the course of our investigations, it became clear that we had
+either failed to consider entirely, or handwaved away as less important
+than they actually were, several key criteria in the name of simplicity.
+The main themes are listed below, as we currently understand them.
+
+### Two way compatibility
+In the [previous
+DR](https://github.com/OpenAssetIO/OpenAssetIO/pull/1254), we attempted
+to make a simplifying assumption based around the idea that managers
+were more capable of managing version updates. Whilst we still believe
+that, it has come to our attention that there are valid reasons that a
+host would also want to maintain backwards compatibility when updating
+versions. Due to this, we now consider the ability for both the host and
+manager to choose to maintain compatibility essential.
+
+### Distributed workflows/RPC
+During investigations, the concept of systems that work remotely, (eg
+[gRPC](https://github.com/foundrytom/OpenAssetIO-gRPC)) came up as a
+potential difficulty with some implementations, given that any callback
+based communication, ie querying the `hostInterface` for a version,
+would be especially difficult to implement in this paradigm.
+We now consider it essential that we take an approach where versioning
+does not impose undue extra effort on distributed workflows.
+
+### No internal state
+OpenAssetIO has long attempted to stick to the philosophy that each
+method call represents a stateless interface, and that the manager
+should not persist state between calls. This is true and essential to
+model potential workflows/integrations in the future (GRPC/HTTP), etc.
+
+However, we do not believe this necessarily excludes any handshake style
+interactions. OpenAssetIO already provides a mechanism to configure
+managers at the beginning of the session with
+[initialize](https://openassetio.github.io/OpenAssetIO/classopenassetio_1_1v1_1_1host_api_1_1_manager.html#aa52c7436ff63ae96e33d7db8d6fd38df).
+Whilst we agree that state must not be utilized during a session,
+injecting version state before or at initialize time remains a valid
+option.
+
+### Naive managers
```

Comment:
This critique only holds if versioning is optional right? In that, in a world where versioning was always required, the distinction between whether a trait does or does not have a version isn't there, you always have to assume there's a version and store it, the only difference being that it's easier if it's in the data because you don't have to do anything extra.

If so, I feel an additional con on out-of-data would convey that better, rather than mentioning that specific behaviour here.